operator-common/src/main/java/io/strimzi/operator/common/ReconciliationException.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common;
+
+import io.strimzi.api.kafka.model.status.Status;
+
+public class ReconciliationException extends Exception {
```

Comment:
What is the purpose behind a new type of exception here?
Just so that a status can be retreieved from it?
Wouldn't it make sense for the reconciler to catch any exceptions and construct the status prior to ending the reconcile?

Reply:
I'm not sure how exactly do you mean it. But we need to be able to pass back the status and yet report that there was an error. The upper class does not have all the info to construct the status. So throwing back just the original exception is not enough and returning success would not properly log the errors and handle metrics.